File: cpp/Program/PerabotanRumahTangga.cpp
#include <bits/stdc++.h>       // Library standar C++
#include "ProdukElektronik.cpp" // Include class induk (ProdukElektronik)
using namespace std;

// ===== Class PerabotanRumahTangga =====
// Turunan dari ProdukElektronik (multilevel inheritance step 1)
// Menambahkan atribut khusus perabotan rumah tangga
class PerabotanRumahTangga : public ProdukElektronik {
private:
    string kategori;  // Kategori perabotan (contoh: Interior, Outdoor, Dekorasi)
    string material;  // Bahan material utama (contoh: Kayu, Plastik, Besi, Kaca)
    float berat;      // Berat produk dalam kilogram

public:
    // Constructor kosong
    PerabotanRumahTangga() {}

    // Constructor dengan parameter (memanggil constructor ProdukElektronik juga)
    PerabotanRumahTangga(string namaProduk, string merek, int harga,
                         string kategori, string material, float berat)
        : ProdukElektronik(namaProduk, merek, harga) {
        this->kategori = kategori;
        this->material = material;
        this->berat = berat;
    }

    // Getter
    string getKategori() { return kategori; }
    string getMaterial() { return material; }
    float getBerat() { return berat; }

    // Setter
    void setKategori(string kategori) { this->kategori = kategori; }
    void setMaterial(string material) { this->material = material; }
    void setBerat(float berat) { this->berat = berat; }

    // Destructor
    ~PerabotanRumahTangga() {}
};

File: php/Program/PerabotanRumahTangga.php
<?php
require_once 'ProdukElektronik.php';

// =============================
// Class PerabotanRumahTangga
// Level 2 (Child dari ProdukElektronik)
// =============================
class PerabotanRumahTangga extends ProdukElektronik {
    private $kategori;   // Kategori produk (misal Interior, Outdoor)
    private $material;   // Material produk (misal Kayu, Plastik)
    private $berat;      // Berat produk (kg)

    public function __construct($namaProduk="", $merek="", $harga=0, $gambar="", $kategori="", $material="", $berat=0.0){
        parent::__construct($namaProduk, $merek, $harga, $gambar);
        $this->kategori = $kategori;
        $this->material = $material;
        $this->berat = $berat;
    }

    // =============================
    // Getter & Setter
    // =============================
    public function getKategori() { return $this->kategori; }
    public function setKategori($kategori) { $this->kategori = $kategori; }

    public function getMaterial() { return $this->material; }
    public function setMaterial($material) { $this->material = $material; }

    public function getBerat() { return $this->berat; }
    public function setBerat($berat) { $this->berat = $berat; }
}
?>